src/helpers.js

```diff
@@ -1,46 +1,85 @@
 const p = require('path')
-const babel = require('babel-core')
-const template = require('babel-template')
-// const printAST = require('ast-pretty-print')
+const fs = require('fs')
 const requireFromString = require('require-from-string')
 
 module.exports = {
+  transformAndRequire,
+  getReplacement,
   replace,
-  looksLike,
+  resolveModuleContents,
   isCodegenComment,
-  getReplacement,
-  stringToAST,
-  resolveModuleToString,
   isPropertyCall,
+  looksLike,
 }
 
-function getReplacement({string: stringToCodegen, filename}) {
-  const {code: transformed} = babel.transform(stringToCodegen, {
+function transformAndRequire({code, filename}, babel) {
+  /**
+   * Transform passed in code to generated code
+   */
+  const {code: transformedCode} = babel.transform(code, {
     filename,
   })
-  const val = requireFromString(transformed, filename)
-  return stringToAST(val)
+
+  /**
+   * Execute the transformed code, as if it were required
+   */
+  const module = requireFromString(transformedCode, filename)
+
+  /**
+   * Allow for es modules (default export)
+   */
+  return module && module.__esModule ? module.default : module
+}
+
+function getReplacement({code, filename, args = [], parserOpts}, babel) {
+  /**
+   * Execute the code string to get the exported module
+   */
+  let module = transformAndRequire({code, filename}, babel)
+
+  /**
+   * If a function is epxorted, call it with args
+   */
+  if (typeof module === 'function') {
+    module = module(...args)
+  } else if (args.length) {
+    throw new Error(
+      `codegen module (${p.relative(
+        process.cwd(),
+        filename,
+      )}) cannot accept arguments because it does not export a function. You passed the arguments: ${args.join(
+        ', ',
+      )}`,
+    )
+  }
+
+  /**
+   * Convert whatever we got now (hopefully a string) into AST form
+   */
+  return codeToAST({code: module, parserOpts}, babel)
 }
 
-function stringToAST(string) {
-  if (typeof string !== 'string') {
+function codeToAST({code, parserOpts = {}}, babel) {
+  if (typeof code !== 'string') {
     throw new Error('codegen: Must module.exports a string.')
   }
-  return template(string, {
+  const parserPlugins = (parserOpts && parserOpts.plugins) || []
+  return babel.template(code, {
     sourceType: 'module',
-    preserveComments: true,
     plugins: [
-      // add more on request...
+      // at least enable a minimum set of plugins
       'jsx',
+      'dynamicImport',
+      'objectRestSpread',
+      ...parserPlugins,
     ],
+    ...parserOpts,
+    preserveComments: true,
+    placeholderPattern: false,
```

Comment:
Wait a second – I got this order wrong, didn't I?
Should be:
```js
{
    preserveComments: true,
    placeholderPattern: false,
    ...parserOpts,
    sourceType: 'module',
    plugins: [
      // at least enable a minimum set of plugins
      'jsx',
      'dynamicImport',
      'objectRestSpread',
      ...parserPlugins,
    ],
  }
```
So sourceType and basePlugins aren't overridable…